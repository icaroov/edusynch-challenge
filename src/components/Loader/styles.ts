import styled, { css } from 'styled-components'

export const Container = styled.div`
  ${({ theme }) => css`
    span {
      background: ${theme.colors.primary};
      width: 5px;
      height: 25px;
      display: inline-block;
      margin: 0 2px;
      border-radius: ${theme.border.radius};
    }

    span:nth-child(1) {
      animation: grow 0.5s -0.45s ease-in-out alternate infinite;
    }
    span:nth-child(2) {
      animation: grow 0.5s -0.3s ease-in-out alternate infinite;
    }
    span:nth-child(3) {
      animation: grow 0.5s -0.15s ease-in-out alternate infinite;
    }
    span:nth-child(4) {
      animation: grow 0.5s 0s ease-in-out alternate infinite;
    }
    @keyframes grow {
      to {
        transform: scaleY(2);
      }
    }
  `}
`
