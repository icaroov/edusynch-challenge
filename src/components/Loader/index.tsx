import * as Styled from './styles'

const Loader = () => (
  <Styled.Container data-testid="loading">
    <span></span>
    <span></span>
    <span></span>
    <span></span>
  </Styled.Container>
)

export default Loader
