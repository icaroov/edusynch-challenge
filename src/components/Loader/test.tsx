import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Loader from '.'

describe('<Loader />', () => {
  it('should render the Loader', () => {
    const { container } = renderWithTheme(<Loader />)

    expect(screen.getByTestId('loading')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
