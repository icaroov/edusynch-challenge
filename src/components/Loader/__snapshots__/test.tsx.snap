// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render the Loader 1`] = `
.c0 span {
  background: #7A57FD;
  width: 5px;
  height: 25px;
  display: inline-block;
  margin: 0 2px;
  border-radius: 0.4rem;
}

.c0 span:nth-child(1) {
  -webkit-animation: grow 0.5s -0.45s ease-in-out alternate infinite;
  animation: grow 0.5s -0.45s ease-in-out alternate infinite;
}

.c0 span:nth-child(2) {
  -webkit-animation: grow 0.5s -0.3s ease-in-out alternate infinite;
  animation: grow 0.5s -0.3s ease-in-out alternate infinite;
}

.c0 span:nth-child(3) {
  -webkit-animation: grow 0.5s -0.15s ease-in-out alternate infinite;
  animation: grow 0.5s -0.15s ease-in-out alternate infinite;
}

.c0 span:nth-child(4) {
  -webkit-animation: grow 0.5s 0s ease-in-out alternate infinite;
  animation: grow 0.5s 0s ease-in-out alternate infinite;
}

<div
  class="c0"
  data-testid="loading"
>
  <span />
  <span />
  <span />
  <span />
</div>
`;
